test/test.spec.js

```diff
@@ -32,3 +37,48 @@ test('removes custom prefix', async t => {
     t.false(f.name.includes('fixtures'))
   })
 })
+
+test('allows read of more files than ulimit maxfiles', async t => {
+  /** @type {string} */
+  let dir
+  try {
+    const totalFiles = 256
+    dir = await generateTestData(totalFiles)
+
+    // Restrict open files to less than the total files we'll read.
+    unlimited(totalFiles - 1)
+
+    const files = await getFilesFromPath(dir)
+    t.is(files.length, totalFiles)
+
+    // Restrict open files to less than the total files we'll read.
+    unlimited(totalFiles - 1)
+
+    // Make sure we can read ALL of these files at the same time.
+    await t.notThrowsAsync(() => Promise.all(files.map(async f => {
+      let i = 0
+      for await (const _ of f.stream()) { // eslint-disable-line no-unused-vars
+        if (i === 0) { // make slow so we open all the files
+          await new Promise(resolve => setTimeout(resolve, 5000))
+        }
+        i++
+      }
+    })))
+  } finally {
+    if (dir) {
+      await promisify(fs.rm)(dir, { recursive: true, force: true })
+    }
+  }
+})
+
+async function generateTestData (n) {
+  const dirName = Path.join(os.tmpdir(), `files-from-path-test-${Date.now()}`)
+  await promisify(fs.mkdir)(dirName)
```

Comment:
cool points but maybe just promisify `fs.rm`, `fs.mkdir`, `fs.writeFile` once at the top